stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,436 @@
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// The module map for the Android NDK.
+// The majority of the modules in this file are prefixed
+// with an underscore, to discourage users from importing them from
+// Swift directly. Instead, users should import 'Android' or another
+// umbrella module that includes these headers.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The top level 'Bionic' module is the C standard library
+// used in the Android NDK.
+module Bionic [system] {
+  header "SwiftBionic.h"
+  export *
+}
+
+// The individual modules for the Bionic C standard library used
+// by the Android NDK.
+module _assert [system] {
+  // <assert.h>'s use of NDEBUG requires textual inclusion.
+  textual header "assert.h"
+}
+module _complex [system] {
+  header "complex.h"
+  export *
+}
+module _ctype [system] {
+  header "ctype.h"
+  export *
+}
+module _errno [system] {
+  header "errno.h"
+  export *
+}
+module _fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers.
+// FIXME: this can be dropped once NDK's libc++ 'std' module is split
+// into individual submodules.
+module _inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use _stdint
+  use _sys_cdefs
+}
+module _limits [system] {
+  header "limits.h"
+  header "bits/posix_limits.h"
+  export *
+}
+module _locale [system] {
+  header "locale.h"
+  export *
+}
+module _malloc [system] {
+  header "malloc.h"
+  export *
+}
+module _math [system] {
+  header "math.h"
+  export *
+  link "m"
+}
+module _pty [system] {
+  header "pty.h"
+  export *
+}
+module _setjmp [system] {
+  header "setjmp.h"
+  export *
+}
+module _signal [system] {
+  header "signal.h"
+  // The 'signal.h' header unfortunately has a circular include
+  // with 'sys/ucontext.h' and thus it must be part of this module.
+  header "sys/ucontext.h"
+  export *
+}
+module _stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+module _stdint [system] {
+  header "stdint.h"
+  export *
+}
+module _stdio [system] {
+  header "stdio.h"
+  export *
+}
+module _stdio_ext [system] {
+  header "stdio_ext.h"
+  export *
+}
+module _stdlib [system] {
+  header "stdlib.h"
+  export *
+}
+module _string [system] {
+  header "string.h"
+  export *
+}
+module _time [system] {
+  header "time.h"
+  export *
+}
+module _utmp [system] {
+  header "utmp.h"
+  header "utmpx.h"
+  export *
+}
+
+// POSIX and another android NDK headers.
+module _alloca [system] {
+  header "alloca.h"
+  export *
+}
+
+module _ar [system] {
+  header "ar.h"
+  export *
+}
+
+module _cpio [system] {
+  header "cpio.h"
+  export *
+}
+
+module _filesystem [system] {
```

Comment:
I was thinking we might give these nicer names like maybe `posix_filesystem` or `Android_filesystem` so that if people want to just get the filesystem parts without paying the cost of compiling the entire Android SDK then they could do that.

Reply:
Or if it's not strictly POSIX we could call it like `filesystem_base` or something to indicate that it's low level fs?